### ISSUE TITLE
Blame view interprets \n as actual newline

### ISSUE THREAD

**Issue by JonathonReinhart** [gitlabhq/gitlabhq#7721]
Gitlab version: 7.2.1

A typical newline escape sequence "\n" will be shown in Blame view as an actual newline:

### Normal view:
![image](https://cloud.githubusercontent.com/assets/1916566/4213387/6655ce70-38b4-11e4-8ba1-7647f57323b9.png)

### Blame view:
![image](https://cloud.githubusercontent.com/assets/1916566/4213406/bfc709ce-38b4-11e4-8bfe-49e0e03fe93c.png)

As you can see in the normal view, the `return 0` is actually on line 6 in the file.  The blame view, however, has interpreted the `\n` sequence of characters as a newline in the file, showing the `return 0` on line 7.

**Comment by JonathonReinhart** [comment]
I don't really know Ruby or Rails, but from what I can tell:
- This problem isn't in the view (`view/projects/blame/show.html/haml`).  It gets the `lines` right from `@blame`:
```
    - @blame.each do |commit, lines, since|
```
- It's not in the controller (`controllers/projects/blame_controller.rb`) either, because it's dead simple:
```
  def show
    @blob = @repository.blob_at(@commit.id, @path)
    @blame = Gitlab::Git::Blame.new(project.repository, @commit.id, @path)
  end
```

So this problem must be in `Gitlab::Git::Blame` itself, which is actually in the [gitlab_git project](https://github.com/gitlabhq/gitlab_git).

**Comment by JonathonReinhart** [comment]
This was fixed somewhere before 7.6.2.
